[FIX] Make excerpt function compatible with multibyte characters

tamalign opened the pull request:
Hi dev team, (@tbantle22)

The current excerpt function works well with half-width characters like those in English, but it does not handle full-width characters like those in Japanese properly.
I have updated the function to check whether characters are half-width or full-width to ensure the truncation works as expected.

Before
![image](https://github.com/dolthub/react-library/assets/712894/3eaa3806-16e0-4486-81e5-dd47106a2ca3)

After
![image](https://github.com/dolthub/react-library/assets/712894/498ca10a-f795-44d6-85a2-3813f83dcf2b)

tbantle22 reviewed (changes requested):
Hi @tamalign, thank you for this contribution! Just ran CI and it looks like `prettier` is failing. This can be resolved by running `yarn prettier-fix` in `utils`. Let me know if you need any help with that

Also do you mind adding some quick unit tests for some full-width characters in [this file](https://github.com/dolthub/react-library/blob/main/packages/utils/src/__tests__/excerpt.test.ts)? Thanks!

tamalign commented:
Hi @tbantle22 ,

Thanks for the review.

I encountered the following error when I moved to `packages/utils` and ran `yarn run prettier-fix` in my environment:
```
error Workspaces can only be enabled in private projects.
```

Since I couldn’t figure out how to resolve this, I temporarily bypassed the issue by adding the private: true flag to package.json. I believe this isn’t the correct approach, so please let me know if you have a proper solution.

I’ve also added unit tests.

Thanks!

tbantle22 commented:
Thanks for making those changes! Looks like lint is failing now

If you run `yarn && yarn ci` in the project root it should run all the tests on all the packages (if you have not used yarn v4 before you may need to run `corepack enable` first as well). You're our first outside contributor and we don't have great documentation for contributors at the moment, but I'll write some next week

Let me know if you have any issues with that. Happy to fix CI and push to this PR if needed

tamalign commented:
After running `corepack enable`, the CI ran successfully 🎉 
Thank you for the support!

tbantle22 commented:
Great! Merging and will get a release out today

tbantle22 commented:
This has been published in @dolthub/web-utils@0.1.5
